File: apps/backend/src/config/env.ts
import dotenv from 'dotenv';

dotenv.config();

export const env = {
  PORT: Number(process.env.PORT) || 8080,
  NODE_ENV: process.env.NODE_ENV || 'development',
  DATABASE_URL: process.env.DATABASE_URL || './data/webui.db',
  JWT_SECRET: process.env.JWT_SECRET || 'change-this-secret-in-production',
  JWT_EXPIRES_IN: process.env.JWT_EXPIRES_IN || '7d',
  CORS_ALLOW_ORIGIN: process.env.CORS_ALLOW_ORIGIN || '*',
  UPLOAD_DIR: process.env.UPLOAD_DIR || './uploads',
  MAX_FILE_SIZE: Number(process.env.MAX_FILE_SIZE) || 10485760, 
  WEBUI_NAME: process.env.WEBUI_NAME || 'LoopLM',
  
  CHROMA_URL: process.env.CHROMA_URL || 'http://localhost:8000',
  CHUNK_SIZE: Number(process.env.CHUNK_SIZE) || 1000,
  CHUNK_OVERLAP: Number(process.env.CHUNK_OVERLAP) || 200,
};

File: apps/backend-python/src/config/env.py
import os
from dotenv import load_dotenv

load_dotenv()


class Env:
    PORT: int = int(os.getenv("PORT", "8080"))
    NODE_ENV: str = os.getenv("NODE_ENV", "development")
    DATABASE_URL: str = os.getenv("DATABASE_URL", "./data/webui.db")
    JWT_SECRET: str = os.getenv("JWT_SECRET", "change-this-secret-in-production")
    JWT_EXPIRES_IN: str = os.getenv("JWT_EXPIRES_IN", "7d")
    CORS_ALLOW_ORIGIN: str = os.getenv("CORS_ALLOW_ORIGIN", "*")
    UPLOAD_DIR: str = os.getenv("UPLOAD_DIR", "./uploads")
    MAX_FILE_SIZE: int = int(os.getenv("MAX_FILE_SIZE", "10485760"))
    WEBUI_NAME: str = os.getenv("WEBUI_NAME", "Wiratek AI")
    # ChromaDB
    CHROMA_MODE: str = os.getenv("CHROMA_MODE", "embedded")  # embedded | http
    CHROMA_PERSIST_DIR: str = os.getenv("CHROMA_PERSIST_DIR", "./data/chroma")
    CHROMA_URL: str = os.getenv("CHROMA_URL", "http://localhost:8000")
    CHUNK_SIZE: int = int(os.getenv("CHUNK_SIZE", "1000"))
    CHUNK_OVERLAP: int = int(os.getenv("CHUNK_OVERLAP", "200"))


env = Env()
